control-plane/config-entries/controllersv2/mesh_gateway_controller.go

```diff
@@ -245,7 +243,7 @@ func (r *MeshGatewayController) getConfigForGatewayClass(ctx context.Context, ga
 	config := &meshv2beta1.GatewayClassConfig{}
 	if ref := gatewayClassConfig.Spec.ParametersRef; ref != nil {
 		if ref.Group != meshv2beta1.MeshGroup ||
-			ref.Kind != common.GatewayClassConfig {
+			ref.Kind != "GatewayClass" {
```

Comment:
```suggestion
			ref.Kind != "GatewayClassConfig" {
```

Reply:
Good call forget we also had to fix it in the other place!